src/lib/parser.js

```diff
@@ -238,16 +359,22 @@ export default function parse(table, config) {
             return;
         }
 
-        parseGames(player, cellsWithResults);
-
         player.tournamentPlace = tournamentPlacement;
-        player.opponents.sort((a, b) => a > b ? 1 : -1);
-
         results[gridPlacement] = player;
 
         lastTournamentPlacement = tournamentPlacement;
         lastGridPlacement = gridPlacement;
     });
 
-    return results;
+    results.forEach(player => {
+        const cells = asArray(player.row.querySelectorAll(settings.cellTags));
+        const cellsWithResults = cells.filter(columnsWithResultsFilter);
+        
+        parseGames(player, cellsWithResults, results, settings);
```

Comment:
I think you can try to move `parseGames` outside the scope of `parse` but keep `cellsWithResults` creation in rows loop (line 298) and use `results` variable to pass result cells. It could allow slimming down `parse` function and keep the main flow simpler.